woocommerce/woocommerce-android
# Update campaign list to support Evergreen campaigns and display CTR

Closes: #12389
This requires these FluxC changes to be merged first: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3086

### Description

This PR adds the following design to Blaze campaign items: 

<img width="300"  src="https://github.com/user-attachments/assets/9d340b00-300c-49bd-a95a-29cd2fa963f0"> 

The logic of how campaign stats are displayed follows these rules: 

- Active (aka `active`, `scheduled` and `InModeration`) campaigns with end date will show “Remaining” budget
- Active Evergreen/Endless campaigns will show  weekly budget
- Finished campaigns will display “Total” spent budget

**EXTRA**: The [prior PR](https://github.com/woocommerce/woocommerce-android/pull/12324) to support Evergreen campaign missed a change to be able to successfully create this type of campaigns. I added that change here as it seemed overkill to create a PR just for it: https://github.com/woocommerce/woocommerce-android/pull/12433/commits/c488eafb631f12c375eae5333d550355de664452

### Testing information
1. Log into a site with Blaze enabled
2. Create an Evergreen campaign and check the budget is displayed as `weekly`
3. Create a campaign with end date and check the budget displays `remaining` while the campaign is active
4. Cancel both campaigns
5. Check Evergreen campaign still displays the `weekly` label
6. Check end date campaigns display `total` label now. 

### The tests that have been performed

I've performed the tests described in the instructions above. 
Smoked tested Blaze campaign creation flow.

### Images/gif


https://github.com/user-attachments/assets/7fb87b6a-208a-4104-9aba-e511eccadec3


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:


12389
